unit_testing/cli/test_cli_sct_register_multimodal.py

```diff
@@ -35,3 +28,36 @@ def test_sct_register_multimodal_mask_no_checks(tmp_path):
         '-m', fname_mask,
         '-initwarp', sct_test_path('t2', 'warp_template2anat.nii.gz')
     ])
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+@pytest.mark.parametrize("use_seg,param,fname_gt", [
+    (False, 'step=1,algo=syn,type=im,iter=1,smooth=1,shrink=2,metric=MI', 'mt/mt0_reg_syn_goldstandard.nii.gz'),
+    (False, 'step=1,algo=slicereg,type=im,iter=5,smooth=0,metric=MeanSquares', 'mt/mt0_reg_slicereg_goldstandard.nii.gz'),
+    (True, 'step=1,algo=centermassrot,type=seg,rot_method=pca', None),
+    (True, 'step=1,algo=centermassrot,type=imseg,rot_method=hog', None),
+    (True, 'step=1,algo=centermassrot,type=imseg,rot_method=pcahog', None),
+    (True, 'step=1,algo=columnwise,type=seg,smooth=1', None),
+])
+def test_sct_register_multimodal_mt0_image_data_within_threshold(use_seg, param, fname_gt):
+    """Run the CLI script and verify that the output image data is close to a reference image (within threshold)."""
+    fname_out = 'mt0_reg.nii.gz'
+
+    argv = ['-i', 'mt/mt0.nii.gz', '-d', 'mt/mt1.nii.gz', '-o', fname_out, '-x', 'linear', '-r', '0', '-param', param]
+    seg_argv = ['-iseg', 'mt/mt0_seg.nii.gz', '-dseg', 'mt/mt1_seg.nii.gz']
+    sct_register_multimodal.main(argv=(argv + seg_argv) if use_seg else argv)
```

Comment:
Is the a chance that  https://github.com/neuropoly/spinalcordtoolbox/issues/3372 was caused by reordering argv?

Reply:
I've written a comment (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3372#issuecomment-861512461) to hopefully help explain the problem a little better.